src/stp/new_plays/BallPlacementUs.cpp

```diff
@@ -0,0 +1,69 @@
+//
+// Created by jessevw on 24.03.20.
+//
+#include "stp/invariants/game_states/BallPlacementUsGameStateInvariant.h"
+#include "stp/new_roles/BallPlacer.h"
+#include "stp/new_roles/Keeper.h"
+#include "stp/new_roles/BallAvoider.h"
+#include "stp/new_plays/BallPlacementUs.h"
+#include "utilities/GameStateManager.hpp"
+
+namespace rtt::ai::stp::play {
+
+    BallPlacementUs::BallPlacementUs() : Play() {
+        startPlayInvariants.clear();
+        startPlayInvariants.emplace_back(std::make_unique<invariant::BallPlacementUsGameStateInvariant>());
+
+        keepPlayInvariants.clear();
+        keepPlayInvariants.emplace_back(std::make_unique<invariant::BallPlacementUsGameStateInvariant>());
+
+        roles = std::array<std::unique_ptr<Role>, rtt::ai::Constants::ROBOT_COUNT()>{
+                std::make_unique<role::BallPlacer>(role::BallPlacer("ball_placer")),    std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_1")),
+                std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_2")),    std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_3")),
+                std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_4")),    std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_5")),
+                std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_6")),    std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_7")),
+                std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_8")),    std::make_unique<role::BallAvoider>(role::BallAvoider("ball_avoider_9")),
+                std::make_unique<role::Keeper>(role::Keeper("keeper"))};
+    }
+
+    uint8_t BallPlacementUs::score(world_new::World* world) noexcept { return 100; }
+
+    void BallPlacementUs::calculateInfoForRoles() noexcept {
+        if (stpInfos.find("ball_placer") != stpInfos.end()) {
+            auto ballTarget = rtt::ai::GameStateManager::getRefereeDesignatedPosition();
+            stpInfos["ball_placer"].setPositionToMoveTo(ballTarget);
+        }
+        if (stpInfos.find("keeper") != stpInfos.end()) {
+            stpInfos["keeper"].setEnemyRobot(world->getWorld()->getRobotClosestToBall(world_new::Team::them));
```

Comment:
I think the keeper also needs a positiontoshootat for it to not segfault

Reply:
Oh okay I will update the issue for this as well